### ISSUE TITLE
Multiplex events not starting if a job is still running

### ISSUE THREAD

**Issue by fl4p** [jhuckaby/Cronicle#175]
## Summary
Cronicle does not start Multiplex events while a job is running on a server.

## Steps to reproduce the problem
* Setup 2 servers running Cronicle
* Create a Multiplex Event on all Servers using Shell plugin and put a `sleep 1h` in the shell script.
* Let the event run and restart Cronicle on the master server
Cronicle will not launch any new jobs for the Event until the job on the slave exited.

## Your Setup
1 Master, 1 Slave

### Operating system and version?
2x Ubuntu 18

### Node.js version?
11

### Cronicle software version?

0.8.28
### Are you using a multi-server setup, or just a single server?
multt
### Are you using the filesystem as back-end storage, or S3/Couchbase?
no
### Can you reproduce the crash consistently?
yes

**Comment by jhuckaby** [comment]
Sounds like you're running into a concurrency limit.  What is your event's [concurrency](https://github.com/jhuckaby/Cronicle#event-concurrency) setting?  If this is set to the default (Singleton) then it will only launch one job for the event at a time.

**Comment by fl4p** [comment]
Yes, you're right the concurrency limit is 1. But isn't that limit for each server?
On a clean start Singleton multiplex event actually spawns one job per server (which is what I want to achieve)

**Comment by jhuckaby** [comment]
Yeah, sorry for the confusion.  Concurrency is global, not per server.  However, the concurrency limit is only checked when the scheduler is about to start a new job.  So even if you have concurrency set to 1, you can still get away with starting a multiplex job targeting multiple servers, but then **all** the jobs have to finish before it starts another.  The best thing in this case is to just set the event concurrency to the total number of servers you are targeting with your multiplex.

**Comment by fl4p** [comment]
Ok, thanks for the clarification. I'm going to try raising the limit then.

So shouldn't the concurrency be treated as a per-server limit for multiplex events? I think a multiplex event that is only allowed to spawn one job globally doesn't make much sense

**Comment by jhuckaby** [comment]
Yeah, I am sorry for the confusion.  Multiplex was really just a last-minute feature / hack that was just sort of tacked on to the system without proper thought.  It's really not designed well at all, and doesn't play nice with concurrency.  It is going to take a redesign to make it work properly.  Maybe for v2.0 😃
